File: src/kyu_7/maximum_length_difference/cpp/maximum_length_difference.cpp
#include <string>
#include <vector>

class MaxDiffLength
{
public:
    static int mxdiflg(std::vector<std::string> &a1, std::vector<std::string> &a2)
    {
        int result = -1;
        for(auto const & a : a1)
        {
            for(auto const & b : a2)
            {
                result = std::max(result, std::abs(int(a.size()) - int(b.size())));
            }
        }
        return result;
    }
};

File: src/kyu_7/maximum_length_difference/csharp/maximum_length_difference.cs
using System;

public class MaxDiffLength
{
    public static int Mxdiflg(string[] a1, string[] a2)
    {
        int result = -1;
            for (int i = 0; i < a1.Length; i++)
            {
                for (int j = 0; j < a2.Length; j++)
                {
                    int currentMax = Math.Abs(a1[i].Length - a2[j].Length);
                    if (currentMax > result)
                    {
                        result = currentMax;
                    }
                }
            }
            return result;
    }
}